File: ios-swiftui/library/Sources/PlanetComponents/TspProgress.swift
import SwiftUI

public struct TspProgress: View {
    let progress: Double
    let variant: String
    let theme: StarPlanetTheme

    public init(progress: Double, variant: String = "primary", theme: StarPlanetTheme = .sky) {
        self.progress = min(max(progress, 0), 100)
        self.variant = variant
        self.theme = theme
    }

    public var body: some View {
        GeometryReader { proxy in
            ZStack(alignment: .leading) {
                Capsule().fill(theme.borderDefault)
                Capsule().fill(fill).frame(width: proxy.size.width * progress / 100)
            }
        }
        .frame(height: 10)
    }

    private var fill: Color {
        switch variant {
        case "success": return theme.success
        case "warning": return theme.warning
        case "danger": return theme.danger
        default: return theme.brandPrimary
        }
    }
}

File: flutter/library/lib/src/tsp_progress.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

class TspProgress extends StatelessWidget {
  const TspProgress({super.key, required this.progress, this.variant = 'primary', this.theme = StarPlanetTheme.sky});
  final double progress;
  final String variant;
  final StarPlanetTheme theme;

  @override
  Widget build(BuildContext context) {
    final fill = switch (variant) {
      'success' => theme.success,
      'warning' => theme.warning,
      'danger' => theme.danger,
      _ => theme.brandPrimary,
    };
    return ClipRRect(
      borderRadius: BorderRadius.circular(999),
      child: SizedBox(
        height: 10,
        child: LinearProgressIndicator(
          minHeight: 10,
          value: (progress / 100).clamp(0, 1),
          backgroundColor: theme.borderDefault,
          color: fill,
        ),
      ),
    );
  }
}
